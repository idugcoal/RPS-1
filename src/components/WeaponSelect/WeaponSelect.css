.weapons {
  display: flex;
  justify-content: center;
  cursor: pointer;
}

.container {
  display: flex;
  flex-direction: column;
  align-items: center;
}
