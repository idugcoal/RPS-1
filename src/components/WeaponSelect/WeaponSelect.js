import React from 'react';
import Weapon from '../Weapon/Weapon';
import { ROCK, PAPER, SCISSORS } from '../../constants';
import style from './WeaponSelect.css';

const WeaponSelect = ({ onClick }) => (
  <div className={style.weaponSelect}>
    <Weapon name='ROCK' onClick={() => onClick(ROCK)} />
    <Weapon name='PAPER' onClick={() => onClick(PAPER)} />
    <Weapon name='SCISSORS' onClick={() => onClick(SCISSORS)} />
  </div>

);

export default WeaponSelect;
