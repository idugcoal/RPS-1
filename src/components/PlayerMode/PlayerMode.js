import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Counter from '../Counter/Counter';
import Battlefield from '../Battlefield/Battlefield';
import WeaponSelect from '../WeaponSelect/WeaponSelect';
import { simulateComputer, computeWinner } from '../../utils';
import style from './PlayerMode.css';

export default class PlayerMode extends Component {
  constructor(props) {
    super(props);
    this.state = {
      playerWins: 0,
      computerWins: 0,
      playerSelection: undefined,
      computerSelection: undefined,
    };
    this.weaponClick = this.weaponClick.bind(this);
    this.reset = this.reset.bind(this);
  }

  reset() {
    this.setState({
      playerWins: 0,
      computerWins: 0,
      playerSelection: undefined,
      computerSelection: undefined,
    });
  }

  weaponClick(weapon) {
    const computerSelection = simulateComputer();
    // 1 means first player wins, -1 means second player wins, 0 is tie
    const result = computeWinner(weapon, computerSelection);
    this.setState(({ playerWins, computerWins }) => ({
      playerSelection: weapon,
      computerSelection,
      playerWins: result === 1 ? playerWins + 1 : playerWins,
      computerWins: result === -1 ? computerWins + 1 : computerWins,
    }));
  }
  render() {
    const {
      playerWins,
      computerWins,
      playerSelection,
      computerSelection,
    } = this.state;
    const { onToggle } = this.props;
    return (
      <div>
        <div className={style.counterContainer}>
          <Counter name='You' wins={playerWins} />
          <div
            className={style.reset}
            onClick={this.reset}
          >
            Reset
          </div>
          <div
            className={style.reset}
            onClick={onToggle}
          >
            Simulation Mode
          </div>
          <Counter name='Computer' wins={computerWins} />
        </div>
        <Battlefield
          firstName='You'
          secondName='Computer'
          firstSelection={playerSelection}
          secondSelection={computerSelection}
        />
        <WeaponSelect onClick={this.weaponClick} />
      </div>
    );
  }
}

PlayerMode.propTypes = {
  onToggle: PropTypes.func,
};
