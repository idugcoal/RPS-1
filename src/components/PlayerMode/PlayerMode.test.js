import React from 'react';
import { shallow } from 'enzyme';
import PlayerMode from './PlayerMode';
import WeaponSelect from '../WeaponSelect/WeaponSelect';

test('should render correctly', () => {
  const wrapper = shallow(<PlayerMode />);
  expect(wrapper).toMatchSnapshot();
});

test('weaponClick should correctly set state', () => {
  const wrapper = shallow(<PlayerMode />);
  wrapper.instance().weaponClick(0);
  expect(wrapper.state().playerSelection).toBe(0);
});

test('reset should correctly reset wins and selections', () => {
  const wrapper = shallow(<PlayerMode />);
  wrapper.instance().weaponClick(0);
  wrapper.instance().weaponClick(1);
  wrapper.instance().reset(0);
  expect(wrapper.state()).toEqual(expect.objectContaining({
    playerWins: 0,
    computerWins: 0,
    playerSelection: undefined,
    computerSelection: undefined,
  }));
});

test('should call weaponClick correctly on weapon select click', () => {
  const wrapper = shallow(<PlayerMode />);
  wrapper.find(WeaponSelect).simulate('click', 0);
  expect(wrapper.state().playerSelection).toBe(0);
});
