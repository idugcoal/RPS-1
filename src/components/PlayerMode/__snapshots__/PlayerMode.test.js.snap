// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <div>
    <ComputerCounter
      name="You"
      wins={0}
    />
    <div
      onClick={[Function]}
    >
      Reset
    </div>
    <div>
      Simulation Mode
    </div>
    <ComputerCounter
      name="Computer"
      wins={0}
    />
  </div>
  <Battlefield
    firstName="You"
    secondName="Computer"
  />
  <WeaponSelect
    onClick={[Function]}
  />
</div>
`;
